src/main/java/scala_maven/ScalaCompilerSupport.java

```diff
@@ -276,6 +289,104 @@ void setLastSuccessfullTS(long v) throws Exception {
     }
   }
 
+  private ScalaInstance getScalaInstanceForS2(
+      File libraryJar,
+      File reflectJar,
+      File compilerJar,
+      VersionNumber scalaVersion,
+      List<File> extraJars)
+      throws Exception {
+
+    List<File> allJars = new ArrayList<>(extraJars);
+    allJars.add(libraryJar);
+    allJars.add(reflectJar);
+    allJars.add(compilerJar);
+
+    URL[] urls =
+        new URL[] {
+          libraryJar.toURI().toURL(), reflectJar.toURI().toURL(), compilerJar.toURI().toURL()
+        };
+
+    URLClassLoader classLoader = new URLClassLoader(urls);
+    return new ScalaInstance(
+        scalaVersion.toString(), // version
+        classLoader, // loader
+        ClasspathUtil.rootLoader(), // loaderLibraryOnly
+        libraryJar, // libraryJar
+        compilerJar, // compilerJar
+        allJars.toArray(new File[] {}), // allJars
+        Option.apply(scalaVersion.toString()) // explicitActual
+        );
+  }
+
+  private ScalaInstance getScalaInstanceForS3(VersionNumber scalaVersion) throws Exception {
+    String version = scalaVersion.toString();
+    String scalaOrg = getScalaOrganization();
+
+    File[] compilerJars =
+        new File[] {
+          getCompilerJar(scalaVersion),
+          getArtifactJar(scalaOrg, SCALA3_INTERFACES_ARTIFACTID, version),
+          getArtifactJar(scalaOrg, "tasty-core_" + version, version),
```

Comment:
This will not work for Scala 3.0.0 onward because the binary version will be "3".
```suggestion
          getArtifactJar(scalaOrg, "tasty-core_" + binaryVersion, version),
```

`binaryVersion` should be `"3.0.0-x"` if Scala version is 3.0.0-x or 3.0.0-x-bin-y
but it should be `"3"` if Scala version >= 3.0.0  (for instance 3.0.0, 3.0.1-RC1, 3.0.1, 3.1.0...)